### ISSUE TITLE
iscan should report 'no access' not 'not registered'

### ISSUE THREAD

**Issue by kript** [irods/irods#3418]
Hi Folks,

We have a zone where the UID of the irods user is different for iCAT as for iRES, and extensive use is made of the ACL's. This means that there are files on the iRES that the local iRODS user doesn't have permission to read.  When you try and verify them with `iscan`, since you do not have access, you actually get a more worrying error message;

```
irods@irods-seq-i21:~$ iscan /irods-seq-i21-bc/fluidigm/e4/18/0f/1662062078/6e/18/ce/S059_1662062078.csv 
/irods-seq-i21-bc/fluidigm/e4/18/0f/1662062078/6e/18/ce/S059_1662062078.csv is not registered in iRODS

irods@irods-seq-i21:~$ ils -l /seq/fluidigm/e4/18/0f/1662062078/6e/18/ce/S059_1662062078.csv 
  srpipe            0 root;replicate;seq-red;red4;irods-seq-i21-bc         2263 2016-11-05.01:30 & S059_1662062078.csv
  srpipe            1 root;replicate;seq-green;green5;irods-seq-sr06-de         2263 2016-11-11.01:37 & S059_1662062078.csv

irods@irods-seq-i21:~$ ils -A /seq/fluidigm/e4/18/0f/1662062078/6e/18/ce/S059_1662062078.csv
  /seq/fluidigm/e4/18/0f/1662062078/6e/18/ce/S059_1662062078.csv
        ACL - ss_4112#seq:read object   trace#Sanger1:read object   dnap_ro#seq:read object   srpipe#Sanger1:own   rodsBoot#seq:own   ds12#Sanger1:read object   hb3#Sanger1:read object   irods-g1#seq:own   psdpipe#Sanger1:read object   kdj#Sanger1:read object 
```

The error should be, I suggest, 'CAT_NO_ACCESS' as would be the case if I tried to `iget` or similar..

**Comment by trel** [comment]
Can you share some more information about the example above?

`whoami`
`ienv`
`ls -l /irods-seq-i21-bc/fluidigm/e4/18/0f/1662062078/6e/18/ce/S059_1662062078.csv`
`ils -L /seq/fluidigm/e4/18/0f/1662062078/6e/18/ce/S059_1662062078.csv`

**Comment by kript** [comment]
```
irods@irods-seq-i21:~$ whoami
irods
irods@irods-seq-i21:~$ ienv
NOTICE: Release Version = rods4.1.9, API Version = d
NOTICE: irods_session_environment_file - /usr/local/iRODS/.irods/irods_environment.json.12165
NOTICE: irods_user_name - irods
NOTICE: irods_host - irods-seq-i21.internal.sanger.ac.uk
NOTICE: xmsg_host is not defined
NOTICE: irods_home - /seq/home/irods
NOTICE: irods_cwd - /seq/home/irods
NOTICE: irods_authentication_scheme is not defined
NOTICE: irods_port - 1247
NOTICE: xmsg_port is not defined
NOTICE: irods_default_resource - irods-seq-i21Resource
NOTICE: irods_zone_name - seq
NOTICE: irods_client_server_policy - CS_NEG_REFUSE
NOTICE: irods_client_server_negotiation - off
NOTICE: irods_encryption_key_size - 32
NOTICE: irods_encryption_salt_size - 8
NOTICE: irods_encryption_num_hash_rounds - 16
NOTICE: irods_encryption_algorithm - AES-256-CBC
NOTICE: irods_default_hash_scheme - MD5
NOTICE: irods_match_hash_policy - compatible
NOTICE: irods_gsi_server_dn is not defined
NOTICE: irods_debug is not defined
NOTICE: irods_log_level is not defined
NOTICE: irods_authentication_file is not defined
NOTICE: irods_ssl_ca_certificate_path is not defined
NOTICE: irods_ssl_ca_certificate_file is not defined
NOTICE: irods_ssl_verify_server is not defined
NOTICE: irods_ssl_certificate_chain_file is not defined
NOTICE: irods_ssl_certificate_key_file is not defined
NOTICE: irods_ssl_dh_params_file is not defined
NOTICE: irods_server_control_plane_key - ZONE_SEQ_CTRL_WELCOMETRUSTSANGER
NOTICE: irods_server_control_plane_encryption_num_hash_rounds - 16
NOTICE: irods_server_control_plane_encryption_algorithm - AES-256-CBC
NOTICE: irods_server_control_plane_port - 1248
NOTICE: irods_maximum_size_for_single_buffer_in_megabytes - 32
NOTICE: irods_default_number_of_transfer_threads - 4
NOTICE: irods_transfer_buffer_size_for_parallel_transfer_in_megabytes - 4
NOTICE: irods_plugins_home is not defined
irods@irods-seq-i21:~$ ls -l /irods-seq-i21-bc/fluidigm/e4/18/0f/1662062078/6e/18/ce/S059_1662062078.csv
-rw------- 1 irods irods 2263 Nov  5 01:30 /irods-seq-i21-bc/fluidigm/e4/18/0f/1662062078/6e/18/ce/S059_1662062078.csv
irods@irods-seq-i21:~$ ils -L /seq/fluidigm/e4/18/0f/1662062078/6e/18/ce/S059_1662062078.csv
  srpipe            0 root;replicate;seq-red;red4;irods-seq-i21-bc         2263 2016-11-05.01:30 & S059_1662062078.csv
    6e18ce4764b1231cade1ece70ecbf48f    generic    /irods-seq-i21-bc/fluidigm/e4/18/0f/1662062078/6e/18/ce/S059_1662062078.csv
  srpipe            1 root;replicate;seq-green;green5;irods-seq-sr06-de         2263 2016-11-11.01:37 & S059_1662062078.csv
    6e18ce4764b1231cade1ece70ecbf48f    generic    /irods-seq-sr06-de/fluidigm/e4/18/0f/1662062078/6e/18/ce/S059_1662062078.csv
```

**Comment by trel** [comment]
So, to make sure I understand the scenario...

If you first add read permission for `irods#seq`:

```
ichmod -M read irods#seq /irods-seq-i21-bc/fluidigm/e4/18/0f/1662062078/6e/18/ce/S059_1662062078.csv
```

Then the `iscan` would return with success?

```
iscan /irods-seq-i21-bc/fluidigm/e4/18/0f/1662062078/6e/18/ce/S059_1662062078.csv
```

If this is true, then I'm still confused about your mentioning different UIDs across different servers.

**Comment by jaking92** [comment]
The fix for this should probably be to confirm existence of the object in question and report "no access" or some permissions error, rather than just "not registered" all the time.

**Comment by jaking92** [comment]
After some discussion, it has been decided that this qualifies as leaking information. This issue will be resolved with the resolution of #3419
